File: basics/TS/src/text/generate.ts
import { BedrockRuntimeClient, InvokeModelCommand } from '@aws-sdk/client-bedrock-runtime'

const client = new BedrockRuntimeClient({ region: 'us-west-2' })

async function invokeModel() {

    const titan_config = {
        inputText: 'Tell me a story about a dragon.',
        textGenerationConfig : {
            maxTokenCount: 4096,
            stopSequences: [],
            temperature: 0,
            topP: 1,
        },
    };
    const titanModelId = 'amazon.titan-text-express-v1';

    const llamaConfig = {
        prompt: 'Generate a story about a jazzman named Vincent and an actress named Mia.',
        // max_gen_len: 512,
        temperature: 0,
        top_p: 0.9,
    }
    const llamaModelId = "meta.llama2-13b-chat-v1"



    const response = await client.send(new InvokeModelCommand({
        body: JSON.stringify(llamaConfig),
        
        modelId: llamaModelId,
        contentType: 'application/json',
        accept: 'application/json',
    }))
    const responseBody = JSON.parse(new TextDecoder().decode(response.body));
    // console.log(responseBody) // titan
    console.log(responseBody.generation) // llama
}
invokeModel()

File: basics/PY/src/text/generate.py
import boto3
import json
import pprint
brt = boto3.client(service_name='bedrock-runtime', region_name="eu-central-1")

anthropic_config = json.dumps({
    "prompt": "\n\nHuman: explain black holes to 8th graders\n\nAssistant:",
    "max_tokens_to_sample": 300,
    "temperature": 0.1,
    "top_p": 0.9,
})

titan_config = json.dumps({
            "inputText": "Tell me a story about a dragon.",
            "textGenerationConfig": {
                "maxTokenCount": 4096,
                "stopSequences": [],
                "temperature": 0,
                "topP": 1
            }
        })

modelId = 'amazon.titan-text-express-v1'
accept = 'application/json'
contentType = 'application/json'

response = brt.invoke_model(body=titan_config, modelId=modelId, accept=accept, contentType=contentType)

response_body = json.loads(response.get('body').read())

pp = pprint.PrettyPrinter(depth=4)

# text
pp.pprint(response_body.get('results'))